vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ComboBoxListDataView.java

```diff
@@ -53,12 +54,38 @@ public ComboBoxListDataView(DataCommunicator<T> dataCommunicator,
         this.dataCommunicator = dataCommunicator;
     }
 
+    /**
+     * Gets the items available on the ComboBox's server side.
+     * <p>
+     * Data is sorted the same way as in the ComboBox, but it does not take
+     * into account the ComboBox client filtering, since it doesn't change
+     * the item count on the server side, but only makes it easier for users
+     * to search through the items in the UI.
+     * Only the server-side filtering considered, which is set by:
+     * {@link #setFilter(SerializablePredicate)} or
+     * {@link #addFilter(SerializablePredicate)}.
+     *
+     * @return filtered and sorted items available in server-side
+     */
     @SuppressWarnings("unchecked")
     @Override
     public Stream<T> getItems() {
         return getDataProvider().fetch(getQuery());
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method takes into account only the server-side filtering, which
+     * is set by:
+     * {@link #setFilter(SerializablePredicate)} or
+     * {@link #addFilter(SerializablePredicate)}.
+     * ComboBox's client filter is not considered, since it doesn't change
```

Comment:
again, maybe change
client -> client side
as one might confuse it otherwise. another option would be "User provided filter from ComboBox's text input is discarded, since..."

Reply:
done